batcher/sweep_batch.go

```diff
@@ -0,0 +1,1109 @@
+
+	// confRefreshChan is a communication channel for the routines that
+	// monitor confirmations. This channel is simply used by the new
+	// routine to signal to the previous routine that it should exit. This
+	// helps the batch have at most one confirmation monitor active at a
+	//
+	confRefreshChan chan int32
+
+	// errChan is the channel over which errors are received.
+	errChan chan error
+
+	// completeChan is the channel over which the batch sends its ID back
+	// to the batcher to signal its exit.
+	completeChan chan lntypes.Hash
+
+	// batchTx is the transaction that is currently being monitored for
+	// confirmations.
+	BatchTxid *chainhash.Hash
+
+	// batchPkScript is the pkScript of the batch transaction's output.
+	BatchPkScript []byte
+
+	// RbfCache stores data related to the RBF fee bumping mechanism.
+	RbfCache RBFCache
+
+	// wallet is the wallet client used to create and publish the batch
+	// transaction.
+	wallet lndclient.WalletKitClient
+
+	// chainNotifier is the chain notifier client used to monitor the
+	// blockchain for spends and confirmations.
+	chainNotifier lndclient.ChainNotifierClient
+
+	// signerClient is the signer client used to sign the batch transaction.
+	signerClient lndclient.SignerClient
+
+	// muSig2Kit includes all the required functionality to collect
+	// and verify signatures by the swap server in order to cooperatively
+	// sweep funds.
+	muSig2SignSweep MuSig2SignSweep
+
+	// store includes all the database interactions that are needed by the
+	// batch.
+	store BatcherStore
+
+	// Cfg is the configuration for this batch.
+	Cfg *BatchConfig
+}
+
+// NewBatch creates a new batch.
+func NewBatch(cfg BatchConfig, batchTxid *chainhash.Hash,
+	returnChan chan *Sweep, completeChan chan lntypes.Hash,
+	wallet lndclient.WalletKitClient,
+	chainNotifier lndclient.ChainNotifierClient,
+	signerClient lndclient.SignerClient, musig2SignSweep MuSig2SignSweep,
+	store BatcherStore) (*Batch, error) {
+
+	newID := lntypes.Hash{}
+	_, err := crand.Read(newID[:])
+	if err != nil {
+		return nil, err
+	}
+
+	return &Batch{
+		ID:               newID,
+		State:            Open,
+		BatchReady:       false,
+		Sweeps:           make(map[lntypes.Hash]Sweep),
+		RequestChan:      make(chan *Sweep),
+		returnChan:       returnChan,
+		blockEpochChan:   make(chan int32),
+		spendChan:        make(chan *chainntnfs.SpendDetail),
+		spendRefreshChan: make(chan int32, 1),
+		confChan:         make(chan *chainntnfs.TxConfirmation),
+		confRefreshChan:  make(chan int32, 1),
+		errChan:          make(chan error),
+		completeChan:     completeChan,
+		BatchTxid:        batchTxid,
+		wallet:           wallet,
+		chainNotifier:    chainNotifier,
+		signerClient:     signerClient,
+		muSig2SignSweep:  musig2SignSweep,
+		store:            store,
+		Cfg:              &cfg,
+	}, nil
+}
+
+// NewBatch creates a new batch that already existed in storage.
+func NewBatchFromDB(cfg BatchConfig, ID lntypes.Hash, batchTxid *chainhash.Hash,
```

Comment:
nit: could use a functional options pattern or a config here to help readability.